### ISSUE TITLE
szf.c:414:2: error: too few arguments to function ‘SZ_batchAddVar’

### ISSUE THREAD

**Issue by jschueller** [disheng222/SZ#30]
The latest release (2.1.5.0) fails to build with :
```
szf.c:414:2: error: too few arguments to function ‘SZ_batchAddVar’
  414 |  SZ_batchAddVar(s2, SZ_FLOAT, data, *errBoundMode, *absErrBound, *relBoundRatio, 0.1, 0, 0, 0, 0, *r1);
      |  ^~~~~~~~~~~~~~
In file included from sz/src/SZ-2.1.5.0/sz/include/sz.h:21,
                 from sz/src/SZ-2.1.5.0/sz/src/szf.c:14:
sz/src/SZ-2.1.5.0/sz/include/VarSet.h:62:6: note: declared here
   62 | void SZ_batchAddVar(int var_id, char* varName, int dataType, void* data,
```

SZ_batchAddVar has one more argument:
```
void SZ_batchAddVar(int var_id, char* varName, int dataType, void* data,
                        int errBoundMode, double absErrBound, double relBoundRatio,
                        double pwRelBoundRatio,
                        size_t r5, size_t r4, size_t r3, size_t r2, size_t r1);
```

**Comment by disheng222** [comment]
Hi Julien,
It seems that you didn't update the SZ code on your side correctly.
We revised the code slightly last month, but the one you showed me is still
old version.
The new code should be like this:
void sz_batchaddvar_d1_float_(int var_id, char* varName, int *len, float*
data, int *errBoundMode, float *absErrBound, float *relBoundRatio, size_t
*r1)
{
int i;
    char s2[*len+1];
    for(i=0;i<*len;i++)
        s2[i]=varName[i];
    s2[*len]='\0';
SZ_batchAddVar(*var_id*, s2, SZ_FLOAT, data, *errBoundMode, *absErrBound,
*relBoundRatio, 0.1, 0, 0, 0, 0, *r1);
}

Please double-check if you pulled the code completely.

Best,
Sheng

**Comment by jschueller** [comment]
You're showing me the git master, As I said I'm using the latest tag (https://github.com/disheng222/SZ/archive/v2.1.5.0.tar.gz), in which the code is different. Please tag a new version, this one is borked.

**Comment by disheng222** [comment]
Thanks for the suggestion.
Done. Just released 2.1.5.1

**Comment by jschueller** [comment]
thanks!
